packages/react-instantsearch-core-next/src/useInstantSearch.ts

```diff
@@ -0,0 +1,31 @@
+import instantsearch from 'instantsearch.js';
+import { useEffect, useMemo, version as ReactVersion } from 'react';
+
+import version from './version';
+
+import type { InstantSearchOptions } from 'instantsearch.js';
+
+export type UseInstantSearchProps = InstantSearchOptions;
+
+export function useInstantSearch(props: UseInstantSearchProps) {
+  const search = useMemo(() => instantsearch(props), [props]);
```

Comment:
same question here, but the props here aren't only strings / stable objects 🤔

Reply:
I already addressed this in the PR that's coming.